### PR TITLE
fixes initial state of gallery and buttons

### LINKED ISSUE #128: DeleteButton not rendered when using addInitialFiles()
I'm having trouble getting the DeleteButton component to work with files loaded using `addInitialFiles()`. I think I've worked out what the issue is but I don't know how to resolve the problem, it may be the way I'm using the components rather than any issue with react-fine-uplaoder.

I'm using the low level components to render a `<Thumbnail />` and a `<DeleteButton />`  component for files added using `addInitialFiles()`. The thumbnail renders ok, and the DeleteButton is generated but the `deletable` state of the `DeleteButton` is set to false which I believe is due to the order of the `statusChange` callbacks.

Here the pertinent sections of my code:

````javascript
componentDidMount() {
  const that = this;
  this.state.uploader.on('statusChange', (id, oldStatus, newStatus) => {
    if (oldStatus === null && newStatus === 'upload successful') {
      // The file was added using addInitialFiles
      that.setState({ files: [{ id, fromServer: true }] });
    } 
  // load the initial files
  this.addInitialFiles();
}

addInitialFiles() {
  this.state.uploader.methods.addInitialFiles(this.props.logos);
}

thumbnail() {
  return (
    this.state.files.map(file => (
      <div key={file.id}>
        <Thumbnail id={file.id} uploader={this.state.uploader} fromServer={file.fromServer} />
        <DeleteButton
          id={file.id}
          uploader={this.state.uploader}
        >
          <i className="fa fa-trash-o" />
        </DeleteButton>
      </div>
      ))
  );
}

render() {
  return (
    <div>
        { this.thumbnail() }
    </div>
  );
}
````

So after my component is mounded I setup a listener that add the file to the `files` state (I am only working with 1 file in this component), it then loads the initial file defined in the `logos` prop. In the render function I'm just looping through the files and rendering a thumb and delete button for each of them.

When a file is added using `addInitialFiles` both components are rendered, but the `DeleteButton` does not show anything as the `deletable` state is set to false. Looking into the `DeleteButton` component code I can see that it sets up it own event listener for `onStatusChange` and will set `deletable` to true if the status is `UPLOAD_SUCCESSFUL`. I believe the trouble is that the `onStatusChange` callback is happening when the file is added using `addInitialFiles` but at this point my code has not rendered the delete button as it is listening to the same event. So when the delete button is rendered the `deletable` status is never changes from false as no more events are fired for this file.

Should I be coding my component differently to avoid this problem? Or is this an issue that needs to be worked around or resolved within react-fine-uploader? I wasn't able to see any way to re-trigger the `onStatusChange` event so that the `DeleteButton` could pick up the correct status of the file.

Thanks

### PR DESCRIPTION
**Pull request by mladen107**
I added initial state based on current state of uploader to galllery, cancel, delete and retry buttons. 

This should fix [#128](https://github.com/FineUploader/react-fine-uploader/issues/128). Also now it is possible to call `uploader.methods.addInitialFiles` before creating gallery and to see initial files.

### CODE REVIEW

**Comment by pyaesone17** [comment]
I think I need deleteable button for initial files.

File: src/cancel-button.jsx [hunk 1]
```diff
@@ -79,6 +80,7 @@ const isCancelable = (statusToCheck, statusEnum) => {
         statusEnum.QUEUED,
         statusEnum.UPLOAD_RETRYING,
         statusEnum.SUBMITTED,
+        statusEnum.SUBMITTING,
```
Comment (rnicholus) [inline src/cancel-button.jsx:83]: Why was this added?

File: src/filename.jsx [hunk 2]
```diff
@@ -23,7 +30,8 @@ class Filename extends Component {
 
     render() {
         return (
-            <span className={ `react-fine-uploader-filename ${this.props.className || ''}` }>
+            <span className={ `react-fine-uploader-filename ${this.props.className || ''}` }
+                  onClick={this._onClick}>
```
Comment (rnicholus) [inline src/filename.jsx:34]: Why does the `<Filename>` component have a click handler? Furthermore, this is completely inaccessible.

**Review by rnicholus — CHANGES_REQUESTED** [review]
Some good changes, but at least a couple should either be removed or fixed.

**Pull request closed without merging**